### DISCUSSION TITLE
ArgumentCompleter to the Task parameter of build.ps1

### DISCUSSION THREAD

**Discussion by webtroter** [devblackops/Stucco#25] [Ideas]
Hi,

So I implemented an ArgumentCompleter for the Task parameter of build.ps1

I'm not sure if it's better in here or if I should also propose it to the PowershellBuild Team.

You can see my commit here : https://github.com/webtroter/Stucco/blob/patch-1/Stucco/template/build.ps1

**Comment by webtroter** [comment]
I'll also show the code here
```powershell
[ArgumentCompleter( {
        param($Command, $Parameter, $WordToComplete, $CommandAst, $FakeBoundParams)
        $psakeFile = './psakeFile.ps1'
        switch ($Parameter) {
            'Task' {
                if ([string]::IsNullOrEmpty($WordToComplete)) {
                    Get-PSakeScriptTasks -buildFile $psakeFile | Select-Object -ExpandProperty Name
                }
                else {
                    Get-PSakeScriptTasks -buildFile $psakeFile |
                        Where-Object { $_.Name -match $WordToComplete } |
                        Select-Object -ExpandProperty Name
                }

            }
            Default {
            }
        }
    })]
```

**Reply by HeyItsGilbert** [reply]
Do you need help making a PR @webtroter? 

@devblackops do you think this would be helpful for folks? I want to say yes because anything that makes development easier for new folks is probably a win.

**Comment by devblackops** [comment]
This looks like a good addition. Thanks!

@webtroter Can you make a PR for this?

**Comment by webtroter** [comment]
Yes I can do a PR. I just wanted some comment about my code and/or integration before.

It works fine with me, but I can't tell for everyone.

**Reply by webtroter** [reply]
I just did one. I didn't remember I had a fork on my profile
https://github.com/devblackops/Stucco/pull/28
